### DISCUSSION TITLE
Rendering issue with for long  reveal.js slides

### DISCUSSION THREAD

**Discussion by Luc-UMass** [quarto-dev/quarto-cli#5458] [Q&A]
Apologies for the somewhat imprecise question.  I am doing slides for my class using reveal.js in Visual Studio Code.   
The slides are fairly long (>200 slides by now) and contains a few snippet of codes in python.  When I hit the 200 pages mark the rendering process has become erratic and often gets stuck. 

I can render easily if I divide my slides in smaller chunks. 

The same issue occur if I try and run my slides using Rstudio. 

Any help/suggestion would be  great.

**Comment by Luc-UMass** [comment]
Maybe this will help.

[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.340
      Path: /Applications/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.1
      Path: /Library/Frameworks/Python.framework/Versions/3.11/bin/python3
      Jupyter: 5.1.3
      Kernels: python3

(/) Checking Jupyter engine render....0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.2.3
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Users/luc/Library/R/arm64/4.2/library
        - /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library
      knitr: 1.41
      rmarkdown: 2.18

[✓] Checking Knitr engine render......OK

**Comment by mcanouil** [comment]
Would you have a repository to share?
Also what do you mean by "run"? Using buttons in VSCode/RStudio? Or actually using `quarto render`?

Slight note, it does not seem very practical to have a "one for all" kind of document.
It would be probably better to create smaller documents (part/section) and include them in a main document.
For example, I have a ~130 slides presentation (https://github.com/mcanouil/radvanced)
May main document looks like this[^1]:
````qmd
---
title: "{{< iconify fa6-brands r-project >}}"
pagetitle: "R - Introduction Avancée"
subtitle: "Introduction Avancée"
author: Mickaël CANOUIL, *Ph.D.*
date: today
date-format: "[Dernière mise à jour : ]YYYY-MM-DD"
lang: fr
execute:
  echo: true
  error: true
  warning: true
embed-resources: true
format:
  coeos-revealjs:
    logo: https://raw.githubusercontent.com/mcanouil/hex-stickers/main/SVG/mc.svg
---

{{< include contents/content-01.qmd >}}
{{< include contents/content-02.qmd >}}
{{< include contents/content-03.qmd >}}
{{< include contents/content-04.qmd >}}
{{< include contents/content-05.qmd >}}
{{< include contents/content-06.qmd >}}
{{< include contents/content-07.qmd >}}
{{< include contents/content-08.qmd >}}
{{< include contents/content-09.qmd >}}
{{< include contents/content-10.qmd >}}
{{< include contents/content-11.qmd >}}
{{< include contents/_about.qmd >}}
````

[^1]: Don't follow my poor file naming in that case, my old self was lazy when splitting the main document.😅
